Sources/StandardLib.swift

```diff
@@ -43,10 +43,12 @@ extension Bool: Decodable {
 }
 
 internal func castOrFail<T>(e: Extractor) throws -> T {
-    let rawValue = e.rawValue
+    return try castOrFail(e.rawValue)
+}
 
-    guard let result = rawValue as? T else {
-        throw typeMismatch("\(T.self)", actual: rawValue, keyPath: nil)
+internal func castOrFail<T>(any: Any?) throws -> T {
```

Comment:
`func castOrFail<T>(any: Any?) throws -> T` is convenient function in `Decodable`, so I would like to it to public function, I think.